express-crud/server.ts

```diff
@@ -0,0 +1,128 @@
+import express from 'express';
+import pg from 'pg';
+import { ClientError, errorMiddleware } from './lib/index.js';
+
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/studentGradeTable',
+  ssl: {
+    // Allow non-SSL traffic to localhost
+    rejectUnauthorized: false,
+  },
+});
+
+const app = express();
+
+app.use(express.json());
+
+app.get('/api/grades', async (req, res, next) => {
+  try {
+    const sql = `
+select *
+from "grades"
+`;
+    const result = await db.query(sql);
+    const gradesRow = result.rows;
+    res.status(200).json(gradesRow);
+  } catch (err) {
+    next(err);
+  }
+});
+
+app.get('/api/grades/:gradeId', async (req, res, next) => {
+  try {
+    const { gradeId } = req.params;
+    if (gradeId === undefined)
+      throw new ClientError(404, 'no grade id found. ');
+    const sql = `
+  select "gradeId","name","course","score"
+  from "grades"`;
```

Comment:
This needs a `where` clause so that it sends back just one specific grade. If you look at the HTTP response carefully on this, you will see that it always returns the first grade instead of the one you requested.